app/models/availability.rb

```diff
@@ -39,4 +41,15 @@ def self.process_data
         { 'Date' => date, 'Count' => count }
       end
   end
+
+  private
+
+  def cancel_duplicates
+    previous_availabilities = person.availabilities.for_time_span(start_time..end_time).order(:created_at)
+    previous_availabilities.each do |a|
+      if start_time == a.start_time && end_time == a.end_time
+        a.update(status: "Cancelled") unless self == a
```

Comment:
This would be a fine first approach. Right now folks have no way to correct a mistake without involving someone who has access to the web UI. The most common use case is an availability for a specific event, so the times would exactly match.